packages/terra-functional-testing/src/reporters/spec-reporter/clean-results.js

```diff
@@ -0,0 +1,25 @@
+const fs = require('fs');
+const path = require('path');
+const getOutputDir = require('./get-output-dir');
+
+/**
+ * Cleans the output directory by removing any previous spec reports.
+ * @param {string} outputDir - The output directory to clean.
+ */
+const cleanResults = (outputDir) => {
+  const directory = outputDir || getOutputDir();
+
+  if (!fs.existsSync(directory)) {
+    return;
+  }
+
+  const files = fs.readdirSync(directory);
```

Comment:
Can we make this async since it's just deleting files?

Reply:
I had written this async originally, but there didn't seem to be many advantages so I ended up swapping it back. At this point in the lifecycle there is only one process and it has to wait for these files to be deleted before moving on to the next lifecycle hook. I'm a little on the fence, I think async in general is better practice , but there are some code complexity trade offs (the hooks will need to await before proceeding)